backend/routes/users.py

```diff
@@ -206,23 +377,30 @@ async def read_user(user_id: UUID, db: AsyncSession = Depends(get_db)):
 
 @router.put("/{user_id}", response_model=schemas.UserRead)
 async def update_user(
-    user_id: UUID, user: schemas.UserBase, db: AsyncSession = Depends(get_db)
+    user_id: UUID,
+    user: schemas.UserBase,
+    db: AsyncSession = Depends(get_db),
+    current_user: models.User = Depends(require_admin_role),
 ):
     """
     Update an existing user's profile information.
 
     This endpoint allows updating user details including username, email,
     and role.
 
+    **Admin Access Required**: Only users with admin role can update users.
+
     Args:
         user_id: The unique identifier of the user to update
         user: Updated user data
         db: Database session dependency
+        current_user: Authenticated admin user (injected by dependency)
 
     Returns:
         schemas.UserRead: The updated user profile
 
     Raises:
+        HTTPException: 403 if the current user is not an admin
```

Comment:
similar to the above `require_admin_role` will raise 403 if the user is not admin

Reply:
same as above.